src/main/kotlin/lotto/domain/WinningLotto.kt

```diff
@@ -1,13 +1,18 @@
 package lotto.domain
 
-class WinningLotto(private val winningNumbers: Set<Int>) {
+class WinningLotto(
+    private val lotto: Lotto,
+    private val bonusNumber: BonusNumber,
+) {
     init {
-        require(winningNumbers.size == 6) { "당첨 번호는 반드시 6개여야 합니다." }
-        require(winningNumbers.all { it in 1..45 }) { "당첨 번호는 1부터 45 사이여야 합니다." }
+        require(bonusNumber.value !in lotto.numbers) {
+            "보너스 번호는 당첨 번호와 중복될 수 없습니다."
+        }
     }
 
-    fun getRank(lotto: Lotto): Rank {
-        val matchCount = lotto.numbers.intersect(winningNumbers).size
-        return Rank.from(matchCount) ?: Rank.MISS
+    fun getRank(lottoToCompare: Lotto): Rank {
+        val matchCount = lottoToCompare.numbers.intersect(lotto.numbers.toSet()).size
+        val matchBonus = bonusNumber.value in lottoToCompare.numbers
```

Comment:
Lotto 클래스를 모두 같이 사용하고 있죠! 그렇다면, 

```suggestion
        val matchCount = lottoToCompare.intersect(lotto).size
        val matchBonus = bonusNumberin lottoToCompare
```

이렇게 객체에서 꺼내 비교하는 것이 아닌, 직접 객체에 물어볼 수 있어 보이는데요. 한번 개선해 보시겠어요?

각 객체에 맞게 역할을 부여하는 거죠!